linked_list/SingleLinkedList.java

```diff
@@ -0,0 +1,107 @@
+////////////////////////////////////////////////////////////////////////////////
+// Author: Ben Kuschner
+////////////////////////////////////////////////////////////////////////////////
+
+public class SingleLinkedList<T> {
```

Comment:
Changed from LinkedList to SingleLinkedList. That way it is very easy to tell this is a single linked list without looking at the implementation.

That way the user knows its limitations and strengths.